### PR TITLE
Andrew/update variable messages

### PR DESCRIPTION
**Pull request by adlondon**
✔︎ I've tested in all browsers (IE11, Firefox, Safari, Chrome, and iOS)

### What should this PR do?
Updates the variable message component

### What is the acceptance criteria?
1. Mimic view of VariableMessage to reflect invision mockup: https://projects.invisionapp.com/share/7GOGNUZEH9V#/screens

[If there are UI changes, include a before and after screenshot.]
image

image

### CODE REVIEW

File: src/less/variable-messages.less [hunk 1]
```diff
@@ -96,10 +89,10 @@
 // 1. Override padding of pill for tighter spacing
 // 2. Help space out stacking variables
 .variable-message__variable {
-  &:extend(.pill);
+  // &:extend(.pill);
```
Comment (benbruning) [inline src/less/variable-messages.less]: Do we want to keep this commented out code?

File: src/less/variable-messages.less [hunk 2]
```diff
@@ -96,10 +89,10 @@
...
   &:extend(.pill--primary);
 
-  padding: 0 @global-whitespace-xsmall; // 1
+  padding: 0; // 1
```
Comment (benbruning) [inline src/less/variable-messages.less]: Same here?

File: src/scripts/components/ToggleButton.jsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    return (
+      <button type="button" className={classes} onClick={this.handleClick} title={this.props.title} {...opts}>
+        <span className="button__text-wrapper">{this.props.children}</span>
+      </button>
+    );
+  }
+}
+
+ToggleButton.propTypes = {
+  available: PropTypes.bool,
```
Comment (benbruning) [inline src/scripts/components/ToggleButton.jsx]: Can any of these be required or all optional?

File: src/scripts/components/VariableMessage.jsx [hunk 4]
```diff
@@ -223,42 +231,32 @@ class VariableMessage extends React.Component {
...
   handleComposeInput = () => {
-    const { variables } = this.props;
     // Get the rawMessage content to return onInput
     const rawMessage = this.compose.textContent.trim();
 
     // Get only the text representation of the message
     // so we can update our DB with it
     let message = rawMessage;
-    const $select = ReactDOM.findDOMNode(this.select);
+    // const $select = ReactDOM.findDOMNode(this.select);
```
Comment (benbruning) [inline src/scripts/components/VariableMessage.jsx]: Keep or leave?

File: src/scripts/components/VariableMessage.jsx [hunk 5]
```diff
@@ -309,11 +310,19 @@ class VariableMessage extends React.Component {
     });
 
     const variableMessageInputName = `variable-message-input-${this.id}`;
-    const variableMessageSelectName = `variable-message-select-${this.id}`;
+    // const variableMessageSelectName = `variable-message-select-${this.id}`;
```
Comment (benbruning) [inline src/scripts/components/VariableMessage.jsx]: Same

File: src/scripts/components/VariableMessage.jsx [hunk 6]
```diff
@@ -115,21 +116,33 @@ class VariableMessage extends React.Component {
...
+  removeVariable = (text) => {
+    const variables = this.getVariables(this.props.variables);
+    const split = this.state.message.split(text).join('').split(/({\w+})/g);
+
+    variables.filter(v => v.value !== text).forEach((value) => {
+      const { variable } = value;
+      const foundVariable = split.indexOf(variable);
+
+      // See if we've found one in our `initialValue`
+      if (foundVariable !== -1) {
```
Comment (alexgsummer) [inline src/scripts/components/VariableMessage.jsx]: ...and now here...
```
if (isVariablePresent)
```

is a bit easier to follow.

File: src/scripts/components/ToggleButton.jsx (hunk 3 above)
Comment (adlondon) [reply src/scripts/components/ToggleButton.jsx]: Can definitely require necessary ones, and I believe I can remove some that are now unnecessary.

**Comment by adlondon** [comment]
Will get better at checking the diffs to spot commented out code. I'll fix these up and resubmit first thing in the am.

**Pull request merged**